src/bad-code.py

```diff
@@ -578,3 +578,60 @@ def Funnel(p,d1,d2):
     
     #sys.stdout=sys.__stdout__
 """
+def create(path,archiveList,xFilesFactor=None,aggregationMethod=None,sparse=False,useFallocate=False):
+  # Set default params
+  if xFilesFactor is None:
+    xFilesFactor = 0.5
+  if aggregationMethod is None:
+    aggregationMethod = 'average'
+
+  #Validate archive configurations...
+  validateArchiveList(archiveList)
+
+  #Looks good, now we create the file and write the header
+  if os.path.exists(path):
+    raise InvalidConfiguration("File %s already exists!" % path)
+  fh = None
+  try:
+    fh = open(path,'wb')
+    if LOCK:
+      fcntl.flock( fh.fileno(), fcntl.LOCK_EX )

```

Comment:
E111 indentation is not a multiple of four
E201 whitespace after '('
E202 whitespace before ')'